lib/apps.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const ApiGroup = require('./api-group');
+
+class Apps extends ApiGroup {
+  constructor(options) {
+    const genericTypes = [
+      // Deprecated name of statefulsets in kubernetes 1.4
+      'petsets',
```

Comment:
Not a bad idea. The only tricky part is that deprecation depends on what k8s version the user is using. We should be showing a support matrix for types and k8s version after this merges